irc/websocket.go

```diff
@@ -0,0 +1,61 @@
+package irc
+
+import (
+	"github.com/gorilla/websocket"
+	"net/http"
+	"time"
+)
+
+var wsUpgrader = websocket.Upgrader{
+	ReadBufferSize:  2 * 1024,
+	WriteBufferSize: 2 * 1024,
+	// If a WS session contains sensitive information, and you choose to use
+	// cookies for authentication (during the HTTP(S) upgrade request), then
+	// you should check that Origin is a domain under your control. If it
+	// isn't, then it is possible for users of your site, visiting a naughty
+	// Origin, to have a WS opened using their credentials. See
+	// http://www.christian-schneider.net/CrossSiteWebSocketHijacking.html#main.
+	// We don't care about Origin because the (IRC) authentication is contained
```

Comment:
Sounds good to me.

Reply:
What about spam? Skipping the origin check means that any site will be able to establish a WebSocket connection, and could use their visitors as distributed spambots.

Reply:
webircgateway includes an "allowed origins" field in their config:

https://github.com/kiwiirc/webircgateway/blob/29955e4a966835be1bef80eba1f1ff116bf0f43d/config.conf.example#L73

This might be the rationale.